src/main/java/hudson/plugins/copyartifact/Copier.java

```diff
@@ -35,7 +35,7 @@
      * @param baseTargetDir Base target dir for upcoming file copy (the copy-artifact
      *   build step may later specify a deeper target dir)
      */
-    public abstract void init(Run src, AbstractBuild<?,?> dst, FilePath srcDir, FilePath baseTargetDir) throws IOException, InterruptedException;
+    public abstract void init(Run src, Run<?,?> dst, FilePath srcDir, FilePath baseTargetDir) throws IOException, InterruptedException;
```

Comment:
https://github.com/jenkinsci/git-plugin/blob/66f632846c21886eef1180daff246d9e0ab973ca/src/main/java/hudson/plugins/git/extensions/GitSCMExtension.java#L65-83 for example.